updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -74,29 +74,33 @@ def save_job_details
 
     def dependency_files_for_multi_directories
       @dependency_files_for_multi_directories ||= job.source.directories.map do |dir|
-        retries = 0
-        begin
+        with_retries do
           updated_dir_source = job.source.clone.tap { |s| s.directory = dir }
           ff = FileFetchers.for_package_manager(job.package_manager).new(
             source: updated_dir_source,
             credentials: Environment.job_definition.fetch("credentials", []),
             options: job.experiments
           )
           ff.files
-        rescue Octokit::BadGateway => e
-          retries += 1
-          retry if retries <= 2
-          raise e
         end
       end.flatten
     end
 
     def dependency_files
-      file_fetcher.files
-    rescue Octokit::BadGateway
-      @file_fetcher_retries ||= 0
-      @file_fetcher_retries += 1
-      @file_fetcher_retries <= 2 ? retry : raise
+      with_retries do
+        file_fetcher.files
+      end
+    end
+
+    def with_retries(max_retries: 2)
+      retries ||= 0
+      begin
+        yield
+      rescue Octokit::BadGateway => e
+        retries += 1
+        retry if retries <= max_retries
+        raise e
```

Comment:
I prefer `raise` to `raise e`. The distinction is quite subtle, `raise` let the original exception be raised, while `raise e` creates a new exception that's a duplicate of the original.

Behavior is essentially the same but you get a less complicated backtrace with `raise`, since with `raise e` you'll get a full backtrace from this location, and then another one from the original exception's location as the exception's cause.

Reply:
Thanks this make more sense. I will make the changes.